infra/src/main/java/com/redhat/service/bridge/infra/models/filters/StringBeginsWith.java

```diff
@@ -2,28 +2,24 @@
 
 import java.util.List;
 
+import javax.validation.constraints.NotEmpty;
+
 import com.fasterxml.jackson.annotation.JsonProperty;
-import com.fasterxml.jackson.core.JsonProcessingException;
-import com.fasterxml.jackson.core.type.TypeReference;
 
 public class StringBeginsWith extends BaseFilter<List<String>> {
     public static final String FILTER_TYPE_NAME = "StringBeginsWith";
 
     @JsonProperty("values")
+    @NotEmpty
     private List<String> values;
 
     public StringBeginsWith() {
         super(FILTER_TYPE_NAME);
     }
 
-    public StringBeginsWith(String key, String values) {
+    public StringBeginsWith(String key, List<String> values) {
         super(FILTER_TYPE_NAME, key);
-        try {
-            this.values = ObjectMapperFactory.get().readValue(values, new TypeReference<List<String>>() {
-            });
-        } catch (JsonProcessingException e) {
-            throw new IllegalArgumentException("The value is not a list of strings.");
```

Comment:
@fjtirado Can we preserve these messages? The reason for the failure is much clearer in the current implementation e.g before your changes the error message I get is: `The value is not a list of strings` versus afterwards, `Type 'StringBeginsWith' is not valid`.

Ideally we'd go to something like `The value of filter type 'StringBeginsWith' must be a list of strings`. I think we were closer with the previous implementation.

Reply:
Current implementation was including in the filter class a responsibility that does not belong to a POJO (build the internal state from a string json representation)  
I would argue that saying that the value that failed is not a list of strings (given the fact that the class that failed is in the stack trace) is more useful that printing that saying that the name of the filter is not valid (former is at least indicating why the code is failing), although I of course agree that we can provide both.